### DISCUSSION TITLE
Time-lapse Mon-Sat, Help with a QR code

### DISCUSSION THREAD

**Discussion by PMyers21** [gopro/labs#757] [Q&A]
Hello, im looking for help with a QR code for a TL on Hero 12.

Im looking to do a 6:00am to 6:00pm long term TL. I need the camera to take a pic every 15mins, Monday through Sat. if possible. If not then it just needs to repeat every day 6am/6pm photo every 15 mins

**Comment by dnewman-gpsw** [comment]
This is the everyday script 
```>06:00<18:00mP+!S+!2N+!900R!06:00NQ!R```

To skip recording on Sunday
```=Wt:W>W3>06:00<18:00mP+!S+!2N+!900R!06:00NQ!R```

The command built in the Blockly interface
![image](https://github.com/gopro/labs/assets/10856084/60a64b5e-96cd-4fc4-af04-6bf6c51eea94)

![Mon-Sat 6am-6pm 15minute Photo-2_7_2024, 09_53_35](https://github.com/gopro/labs/assets/10856084/a7d1c325-5505-41fc-99a1-c91285c4ddd0)

**Reply by PMyers21** [reply]
Thanks so much! Any chance you have that Blocky build on the everyday 6-6 TL. 

Also, how could i make a 24hr, every 15min TL? Is there somewhere to learn that Blocky interface easily?

**Reply by dnewman-gpsw** [reply]
Save the PNG with QR code from above to you computer.  Then open the block interface https://gopro.github.io/labs/build/ then click on Import Image, give it the PNG.  Now just play to learn.

**Comment by PMyers21** [comment]
How do you save the QR codes like that if you insert a script. I tried the labs one but when I import to the build software it says its missing an XML file.

**Reply by dnewman-gpsw** [reply]
Right click on image, select "Save link as" or "Save image as", should work in Chrome or Brave browsers.  The file will be something like 303088851-a7d1c325-5505-41fc-99a1-c91285c4ddd0.png, you can rename it keeping the .png.  The metadata is in the PNG. Working on all three browsers I have.

P.S. This will only work with images generated with Labs Blockly.  The Images will always have this header
![Clipboard01 jog](https://github.com/gopro/labs/assets/10856084/4b8f574c-128b-4ef1-ac7c-7de35bbb5d68)

**Comment by PMyers21** [comment]
Ya it works when importing the .png

I was trying to mess around with this code   >06:00<18:00mP+!S+!2N+!900R!06:00NQ!R  and try to make it a 24hr every 5 min TL

**Reply by dnewman-gpsw** [reply]
24HR 5 minutes is just ```!S!300RQ```

**Comment by gilmbento** [comment]
> This is the everyday script `>06:00<18:00mP+!S+!2N+!900R!06:00NQ!R`
> 
> To skip recording on Sunday `=Wt:W>W3>06:00<18:00mP+!S+!2N+!900R!06:00NQ!R`
> 
> The command built in the Blockly interface ![image](https://private-user-images.githubusercontent.com/10856084/303088633-60a64b5e-96cd-4fc4-af04-6bf6c51eea94.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQxMjQyOTcsIm5iZiI6MTcxNDEyMzk5NywicGF0aCI6Ii8xMDg1NjA4NC8zMDMwODg2MzMtNjBhNjRiNWUtOTZjZC00ZmM0LWFmMDQtNmJmNmM1MWVlYTk0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI2VDA5MzMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjOWYzMmNkMGIwMzM4MGMxOTQyYjU2YjFhODA1ZjA0ZjYwZjAxMmYzMGUwNjZhMTRlZDAxNzU1M2Y4NDlhNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RT-oERE9UAG-pJvSMsLujTFZhvwi7sxg5GUr9PS2hDA)
> 

Is it possible to adapt this script to skip the whole weekend ?
The same behaviour but from monday to friday ?

Thank you very much.

**Reply by dnewman-gpsw** [reply]
Yes.  Do you have any coding skills?  In the example above variable W has the day of the week.  Mon-Fri will have have W between 1 and 5. Just set the condition for that range.

**Reply by gilmbento** [reply]
No, i dont have coding skills but i think i undestood the logic .
In this case , the code would start like this : =Wt:W>W1<W6 ?

Thank you !

**Reply by dnewman-gpsw** [reply]
You can have the level of coding skills required.  ;)

**Reply by HotteXY** [reply]
Please excuse me for bringing this topic up again.

>06:00<18:00mP+!S+!2N+!900R!06:00NQ!R

I don't understand where the + comes from. 
P+ for example 
I can't find such a command. Is this caused by Blockly? 
Otherwise, thank you for this attitude. Has been working absolutely reliably on the Hero 9 for 4 days.

Greetings Mario

**Reply by dnewman-gpsw** [reply]
this is if statement in the equivalent C code.

```
if(time > 6 && time < 18)
{
  modePhoto();
  Start();
  Sleep(2);
  Repeat(900);
}
```
or
```
if(time > 6 && time < 18) {  modePhoto();  Start();  Sleep(2);  Repeat(900);  }
```

C doesn't care about spacing or linefeeds.

Those braces are very useful, to separate out different operations that happen when the condition is true, but the Labs command language doesn't have braces.

However the same C code can be written without braces this way:
```
if(time > 6 && time < 18) modePhoto(), Start(), Sleep(2), Repeat(900);
```

Here the comma is saying, "and do this", "and do that", etc.

This is what the Labs "+" is doing.  The plus is for the following commands to be part of the previous conditional.

Try this:
```<12:00"Good"+"Morning"~"Good"+"Afternoon"```


As for `P+` is not command, true, it is not, but `mP` is.  All command starts with a lower case character: r4X, mTL, fW0, etc. or there start with ! for an action, or < > =, etc. conditional or assignment.  Nothing starts with upper case.

**Reply by HotteXY** [reply]
Thank you for the detailed explanation. I'll try that out. 
I'm really impressed with your skills. 
It's great that you share it with us.
